spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -246,101 +288,152 @@ <h1>GetStringIndex ( _S_, _Input_, _e_ )</h1>
-            1. <del>Perform ? Set(_rx_, `"lastIndex"`, _nextIndex_, *true*).</del>
-          1. <ins>If IsCallable(_mapfn_) is *true*, then</ins>
-            1. <ins>Let _matchValue_ be ? Get(_match_, `"0"`).</ins>
-          1. <ins>Else,</ins>
-            1. <ins>Let _matchValue_ be ? ToString(? Get(_match_, `"0"`)).</ins>
+          1. <del>Let _matchStr_ be ? ToString(? Get(_result_, `"0"`)).</del>
+          1. <ins>Let _matchValue_ be ? CoerceMatchResult( ? Get(_result_, "0"), _matchOptions_).</ins>
           1. Let _status_ be CreateDataProperty(_A_, ! ToString(_n_), <del>_matchStr_</del><ins>_matchValue_</ins>).
           1. Assert: _status_ is *true*.
-          1. <ins>Let _thisIndex_ be ? ToLength(? Get(_rx_, `"lastIndex"`)).</ins>
-          1. <ins>Let _matchLength_ be _thisIndex_-_lastIndex_.</ins>
-          1. <ins>If _matchLength_&le;0, then</ins>
-            1. <ins>Let _nextIndex_ be AdvanceStringIndex(_S_, _thisIndex_, _fullUnicode_).</ins>
-            1. <ins>Perform ? Set(_rx_, `"lastIndex"`, _nextIndex_, *true*).</ins>
-            1. <ins>Set _lastIndex_ to _nextIndex_.</ins>
-          1. <ins>Else, set _lastIndex_ to _thisIndex_.</ins>
+          1. <del>If _matchStr_ is the empty String, then</del>
+          1. <ins>If ? IsEmptyMatch(_rx_, _matchOptions_) is *true*, then</ins>
+            1. Let _thisIndex_ be ? ToLength(? Get(_rx_, `"lastIndex"`)).
+            1. Let _nextIndex_ be AdvanceStringIndex(_S_, _thisIndex_, _fullUnicode_).
+            1. Perform ? Set(_rx_, `"lastIndex"`, _nextIndex_, *true*).
           1. Set _n_ to _n_ + 1.
     </emu-alg>
     <p>The value of the `name` property of this function is `"[Symbol.match]"`.</p>
     <emu-note>
       <p>The @@match property is used by the IsRegExp abstract operation to identify objects that have the basic behaviour of regular expressions. The absence of a @@match property or the existence of such a property whose value does not Boolean coerce to *true* indicates that the object is not intended to be used as a regular expression object.</p>
     </emu-note>
+
   </emu-clause>
 
-  <emu-clause id="sec-regexp-prototype-matchall">
-    <h1>RegExp.prototype [ @@matchAll ] ( _string_ <ins>[, _mapfn_]</ins>)</h1>
-    <p>When the `@@matchAll` method is called with argument _string_ <ins> and optional argument _mapfn_</ins>, the following steps are taken:</p>
+  <emu-clause id="sec-regexp-prototype-@@matchall">
+    <h1>RegExp.prototype [ @@matchAll ] ( _string_ <ins>[, _options_]</ins>)</h1>
+    <p>When the `@@matchAll` method is called with argument _string_ <ins> and optional argument _options_</ins>, the following steps are taken:</p>
     <emu-alg>
     1. Let _R_ be the *this* value.
     1. If Type(_R_) is not Object, throw a *TypeError* exception.
     1. Let _S_ be ? ToString(_string_).
+    1. <ins>If _options_ is not present, let _options be *undefined*.</ins>
```

Comment:
this line is not required; all optional arguments always default to undefined when not present.
```suggestion
```